Temporary fix for:

dyld: lazy symbol binding failed: Symbol not found: _SCDynamicStoreCopyProxies
  Referenced from: /opt/local/lib/libcurl.4.dylib
  Expected in: flat namespace

due to failure to properly identify macOS when using the macOS 10.11 SDK or
earlier. This patch is only tested on macOS and iOS. The upstream fix, which
needs to accommodate iOS, tvOS, watchOS, and every other OS may be different.

https://github.com/curl/curl/issues/11502
https://github.com/curl/curl/pull/11516
--- configure.orig	2023-07-23 16:00:43.000000000 -0500
+++ configure	2023-07-27 00:57:35.000000000 -0500
@@ -21707,7 +21707,7 @@
 int main (void)
 {
 
-#if (TARGET_OS_OSX)
+#if TARGET_OS_MAC && !(defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)
       return 0;
 #else
 #error Not a macOS
--- lib/curl_setup.h.orig	2023-07-23 10:14:56.000000000 -0500
+++ lib/curl_setup.h	2023-07-27 00:53:49.000000000 -0500
@@ -258,8 +258,9 @@
 #if defined(__APPLE__) && !defined(USE_ARES)
 #include <TargetConditionals.h>
 #define USE_RESOLVE_ON_IPS 1
-#  if !defined(TARGET_OS_OSX) || TARGET_OS_OSX
-#    define CURL_OSX_CALL_COPYPROXIES 1
+#  if TARGET_OS_MAC && !(defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE) && \
+     defined(ENABLE_IPV6)
+#    define CURL_MACOS_CALL_COPYPROXIES 1
 #  endif
 #endif
 
--- lib/macos.c.orig	2023-07-18 04:46:48.000000000 -0500
+++ lib/macos.c	2023-07-27 00:55:43.000000000 -0500
@@ -26,19 +26,16 @@
 
 #if defined(__APPLE__)
 
-#if !defined(TARGET_OS_OSX) || TARGET_OS_OSX
+#ifdef CURL_MACOS_CALL_COPYPROXIES
 
 #include <curl/curl.h>
 
 #include "macos.h"
 
-#if defined(ENABLE_IPV6) && defined(CURL_OSX_CALL_COPYPROXIES)
 #include <SystemConfiguration/SCDynamicStoreCopySpecific.h>
-#endif
 
 CURLcode Curl_macos_init(void)
 {
-#if defined(ENABLE_IPV6) && defined(CURL_OSX_CALL_COPYPROXIES)
   {
     /*
      * The automagic conversion from IPv4 literals to IPv6 literals only
@@ -53,7 +50,6 @@
     if(dict)
       CFRelease(dict);
   }
-#endif
   return CURLE_OK;
 }
 
--- lib/macos.h.orig	2023-07-18 04:46:48.000000000 -0500
+++ lib/macos.h	2023-07-27 00:56:07.000000000 -0500
@@ -25,7 +25,7 @@
  ***************************************************************************/
 #include "curl_setup.h"
 
-#if defined(__APPLE__) && (!defined(TARGET_OS_OSX) || TARGET_OS_OSX)
+#ifdef CURL_MACOS_CALL_COPYPROXIES
 
 CURLcode Curl_macos_init(void);
 
